CMakeLists.txt

```diff
@@ -1,332 +1,387 @@
+deprecated_option(OLD NGEN_ACTIVATE_ROUTING NEW NGEN_WITH_ROUTING)
+deprecated_option(OLD NETCDF_ACTIVE         NEW NGEN_WITH_NETCDF)
+deprecated_option(OLD BMI_C_LIB_ACTIVE      NEW NGEN_WITH_BMI_C)
+deprecated_option(OLD BMI_FORTRAN_ACTIVE    NEW NGEN_WITH_BMI_FORTRAN)
+deprecated_option(OLD QUIET                 NEW NGEN_QUIET)
+deprecated_option(OLD PACKAGE_TESTS         NEW NGEN_WITH_TESTS)
+# Project =====================================================================
+
+# -----------------------------------------------------------------------------
+# Set global C++ options
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+
+# Define project version and use via generated config header
 project(ngen VERSION 0.1.0)
-configure_file(include/NGenConfig.h.in include/NGenConfig.h)
 
-if("${PROJECT_NAME}" STREQUAL "${CMAKE_PROJECT_NAME}")
-    option(PACKAGE_TESTS "Build automated tests" ON)
+configure_file("${NGEN_INC_DIR}/NGenConfig.h.in" "${NGEN_INC_DIR}/NGenConfig.h")
+
+add_executable(ngen "${NGEN_SRC_DIR}/NGen.cpp")
+
+# Dependencies ================================================================
+
+# -----------------------------------------------------------------------------
+# Check if NGen is the main project (e.g. development environment)
+if(PROJECT_NAME STREQUAL CMAKE_PROJECT_NAME)
+    set(NGEN_IS_MAIN_PROJECT ON)
 else()
-    option(PACKAGE_TESTS "Build automated tests")
+    set(NGEN_IS_MAIN_PROJECT OFF)
 endif()
 
+# -----------------------------------------------------------------------------
+# If MPI support is enabled, set the C++ compiler to "mpicxx"
+if(NGEN_WITH_MPI)
+    find_package(MPI REQUIRED)
+    add_compile_definitions(NGEN_MPI_ACTIVE)
+endif()
+
+# -----------------------------------------------------------------------------
+# FIXME: Is this needed for icpx? or even icpp? This only enables support for SYCL - Justin
 # https://www.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top/compiler-setup/use-the-command-line/use-cmake-with-the-compiler.html
-if (INTEL_DPCPP)
+if(INTEL_DPCPP)
     find_package(IntelDPCPP REQUIRED)
 endif()
 
-set(CMAKE_CXX_STANDARD 14)
-set(CMAKE_CXX_STANDARD_REQUIRED True)
-set(CMAKE_CXX_VISIBILITY_PRESET hidden)
-
-include(ExternalProject)
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/modules/")
-
+# -----------------------------------------------------------------------------
 # Account for different OS and how that impacts shared lib file names
-if ("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
+# FIXME: What is this used for? - Justin
+if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
     set(SHARED_LIB_EXTENSION "dylib")
 elseif(WIN32)
-    error( "Windows platforms are not currently supported" )
+    error("Windows platforms are not currently supported")
 else()
     set(SHARED_LIB_EXTENSION "so")
 endif()
```

Comment:
This entire block is unused now - it was only used for the bundled UDUNITS2 build, which was eliminated in #561.

It maybe should inform the code in https://github.com/NOAA-OWP/ngen/blob/master/include/realizations/catchment/AbstractCLibBmiAdapter.hpp#L80 but it doesn't currently.

Reply:
I've changed the name of the variable to `NGEN_SHARED_LIB_EXTENSION` and added it as a compile definition. No changes to the dynamic loader, but it's setup for that now, which I think should be useful.